crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -174,6 +175,19 @@ pub struct Res<'w, T: Component> {
     change_tick: u32,
 }
 
+impl<'w, T: Component> Debug for Res<'w, T>
+where
+    T: Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("Res(")?;
+        Debug::fmt(self.value, f)?;
+        f.write_str(")")?;
+
+        Ok(())
```

Comment:
These (and all other `Debug` impls) can all be reduce to a single line:
```rust
f.write_fmt(format_args!("Res({:?})", self.value))
```

Reply:
Also, when you `#[derive(Debug)]` for a struct, normally the format is: `StructName { <values> }`

So perhaps we should stick with this style, unless we want to explicitly mimic a tuple-like formatting.

Tuple-like:
```rust
"Res(<value>)"
```

Struct-like:
```rust
"Res { value: <value> }"
```

Reply:
Changed all the implementations to use this method.

Reply:
rust debug actually has helper for that https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_tuple for tuple
```
        fmt.debug_tuple("Res")
            .field(&self.value)
            .finish()
```
or https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct for struct

they let caller control more of the formatting, like precision, pretty print, ...